Remote EJB lookup failed after upgrade from 171 to 172

Hi,

After upgrade from v171 to v172 community stream version, an error occured if we lookup from backend clustered system to the frontend clustered system (with unchanged applications and configurations).
Other local type lookups work well.

Backend EJB configuration files:
```
<ejb-jar>
  <enterprise-beans>
    ...
    <session>
      <ejb-name>BackendSFBean</ejb-name>
      <transaction-type>Container</transaction-type>
      <ejb-ref>
        <ejb-ref-name>ejb/FrontendSFBean</ejb-ref-name>
        <ejb-ref-type>Session</ejb-ref-type>
        <remote>mycompany.sf.FrontendSFRemote</remote>
      </ejb-ref>
    </session>
    ...
  </enterprise-beans>
</ejb-jar>

<glassfish-ejb-jar>
  <enterprise-beans>
    ...
    <ejb>
      <ejb-name>BackendSFBean</ejb-name>
      <jndi-name>ejb/BackendSFBean</jndi-name>
      <ejb-ref>
        <ejb-ref-name>ejb/FrontendSFBean</ejb-ref-name>
        <jndi-name>corbaname:iiop:ejbfrontendhost1:33700,iiop:ejbfrontendhost2:33700#ejb/FrontendSFBean</jndi-name>
      </ejb-ref>
    </ejb>
    ...
  </enterprise-beans>
</glassfish-ejb-jar>
```
backend code:
```
javax.naming.Context ctx = new javax.naming.InitialContext();
FrontendSFRemote remote = (FrontendSFRemote)ctx.lookup("java:comp/env/ejb/FrontendSFBean"); //this JNDI subcontext originally from external properties file
```
server.log:
```
...
javax.naming.NamingException: Lookup failed for 'java:comp/env/ejb/FrontendSFBean' in SerialContext[myEnv={java.naming.factory.initial=com.sun.enterprise.naming.impl.SerialInitContextFactory, java.naming.factory.state=com.sun.corba.ee.impl.presentation.rmi.JNDIStateFactoryImpl, java.naming.factory.url.pkgs=com.sun.enterprise.naming} [Root exception is javax.naming.NamingException: Exception resolving Ejb for 'Remote ejb-ref name=ejb/FrontendSFBean,Remote 3.x interface =mycompany.sf.FrontendSFRemote,ejb-link=null,lookup=,mappedName=,jndi-name=corbaname:iiop:ejbfrontendhost1:33700,iiop:ejbfrontendhost2:33700#ejb/FrontendSFBean,refType=Session' .  Actual (possibly internal) Remote JNDI name used for lookup is 'corbaname:iiop:ejbfrontendhost1:33700,iiop:ejbfrontendhost2:33700#ejb/FrontendSFBean__3_x_Internal_RemoteBusinessHome__' [Root exception is org.omg.CORBA.BAD_PARAM: FINE: 00100009: string_to_object conversion failed due to bad schema specific part in name ejb/FrontendSFBean__3_x_Internal_RemoteBusinessHome__  vmcid: SUN  minor code: 9  completed: No]]
	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:494)
	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:440)
	at javax.naming.InitialContext.lookup(InitialContext.java:417)
	at javax.naming.InitialContext.lookup(InitialContext.java:417)
	at mycompany.sf.BackendSFBean.anyBackendMethod(BackendSFBean.java:9640)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.glassfish.ejb.security.application.EJBSecurityManager.runMethod(EJBSecurityManager.java:1081)
	at org.glassfish.ejb.security.application.EJBSecurityManager.invoke(EJBSecurityManager.java:1153)
	at com.sun.ejb.containers.BaseContainer.invokeBeanMethod(BaseContainer.java:4762)
	at com.sun.ejb.EjbInvocation.invokeBeanMethod(EjbInvocation.java:656)
	at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:834)
	at com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:608)
	at mycompany.interceptor.AuditInterceptor.logMethodCall(AuditInterceptor.java:38)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:895)
	at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:834)
	at com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:608)
	at org.jboss.weld.ejb.AbstractEJBRequestScopeActivationInterceptor.aroundInvoke(AbstractEJBRequestScopeActivationInterceptor.java:73)
	at org.jboss.weld.ejb.SessionBeanInterceptor.aroundInvoke(SessionBeanInterceptor.java:52)
	at sun.reflect.GeneratedMethodAccessor138.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:895)
	at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:834)
	at com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:608)
	at com.sun.ejb.containers.interceptors.SystemInterceptorProxy.doCall(SystemInterceptorProxy.java:163)
	at com.sun.ejb.containers.interceptors.SystemInterceptorProxy.aroundInvoke(SystemInterceptorProxy.java:140)
	at sun.reflect.GeneratedMethodAccessor92.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:895)
	at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:834)
	at com.sun.ejb.containers.interceptors.InterceptorManager.intercept(InterceptorManager.java:374)
	at com.sun.ejb.containers.BaseContainer.__intercept(BaseContainer.java:4734)
	at com.sun.ejb.containers.BaseContainer.intercept(BaseContainer.java:4722)
	at com.sun.ejb.containers.EJBObjectInvocationHandler.invoke(EJBObjectInvocationHandler.java:205)
	at com.sun.ejb.containers.EJBObjectInvocationHandler.invoke(EJBObjectInvocationHandler.java:123)
	at com.sun.proxy.$Proxy538.anyBackendMethod(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie.dispatchToMethod(ReflectiveTie.java:143)
	at com.sun.corba.ee.impl.presentation.rmi.ReflectiveTie._invoke(ReflectiveTie.java:173)
	at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatchToServant(ServerRequestDispatcherImpl.java:528)
	at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:199)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1564)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:1440)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleInput(MessageMediatorImpl.java:945)
	at com.sun.corba.ee.impl.protocol.giopmsgheaders.RequestMessage_1_2.callback(RequestMessage_1_2.java:213)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:709)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.dispatch(MessageMediatorImpl.java:508)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.doWork(MessageMediatorImpl.java:2237)
	at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$TaskRunner.run(ThreadPoolImpl.java:178)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: javax.naming.NamingException: Exception resolving Ejb for 'Remote ejb-ref name=ejb/FrontendSFBean,Remote 3.x interface =mycompany.sf.FrontendSFRemote,ejb-link=null,lookup=,mappedName=,jndi-name=corbaname:iiop:ejbfrontendhost1:33700,iiop:ejbfrontendhost2:33700#ejb/FrontendSFBean,refType=Session' .  Actual (possibly internal) Remote JNDI name used for lookup is 'corbaname:iiop:ejbfrontendhost1:33700,iiop:ejbfrontendhost2:33700#ejb/FrontendSFBean__3_x_Internal_RemoteBusinessHome__' [Root exception is org.omg.CORBA.BAD_PARAM: FINE: 00100009: string_to_object conversion failed due to bad schema specific part in name ejb/FrontendSFBean__3_x_Internal_RemoteBusinessHome__  vmcid: SUN  minor code: 9  completed: No]
	at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:188)
	at com.sun.enterprise.container.common.impl.ComponentEnvManagerImpl$EjbReferenceProxy.create(ComponentEnvManagerImpl.java:1015)
	at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:745)
	at com.sun.enterprise.naming.impl.GlassfishNamingManagerImpl.lookup(GlassfishNamingManagerImpl.java:715)
	at com.sun.enterprise.naming.impl.JavaURLContext.lookup(JavaURLContext.java:159)
	at com.sun.enterprise.naming.impl.SerialContext.lookup(SerialContext.java:474)
	... 67 more
Caused by: org.omg.CORBA.BAD_PARAM: FINE: 00100009: string_to_object conversion failed due to bad schema specific part in name ejb/FrontendSFBean__3_x_Internal_RemoteBusinessHome__  vmcid: SUN  minor code: 9  completed: No
	at com.sun.proxy.$Proxy227.soBadSchemaSpecific(Unknown Source)
	at com.sun.corba.ee.impl.resolver.INSURLOperationImpl.resolveCorbaname(INSURLOperationImpl.java:227)
	at com.sun.corba.ee.impl.resolver.INSURLOperationImpl.resolveINSURL(INSURLOperationImpl.java:154)
	at com.sun.corba.ee.impl.resolver.INSURLOperationImpl.operate(INSURLOperationImpl.java:145)
	at com.sun.corba.ee.impl.orb.ORBImpl.string_to_object(ORBImpl.java:977)
	at com.sun.ejb.EjbNamingReferenceManagerImpl.resolveEjbReference(EjbNamingReferenceManagerImpl.java:181)
	... 72 more
Caused by: org.omg.CosNaming.NamingContextPackage.NotFound: IDL:omg.org/CosNaming/NamingContext/NotFound:1.0
	at org.omg.CosNaming.NamingContextPackage.NotFoundHelper.read(NotFoundHelper.java:72)
	at org.omg.CosNaming._NamingContextExtStub.resolve_str(_NamingContextExtStub.java:165)
	at com.sun.corba.ee.impl.resolver.INSURLOperationImpl.resolveCorbaname(INSURLOperationImpl.java:222)
	... 76 more
```

## Environment ##

- **Payara Version**: 4.1.1.172
- **Edition**: Full
- **JDK Version**: 8 u131 Oracle
- **Operating System**: Windows

--- a/nucleus/pom.xml
+++ b/nucleus/pom.xml
@@ -162,7 +162,7 @@
         <findbugs.version>2.5.2</findbugs.version>
         <findbugs.glassfish.logging.validLoggerPrefixes>javax.enterprise</findbugs.glassfish.logging.validLoggerPrefixes>
 
-        <glassfish-corba.version>4.0.1.payara-p7</glassfish-corba.version>
+        <glassfish-corba.version>4.0.1.payara-p8</glassfish-corba.version>
         <stax-api.version>1.0-2</stax-api.version>
         <management-api.version>1.1-rev-1</management-api.version>
         <servlet-api.version>3.1.0</servlet-api.version>

